docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,170 @@
+
+This normalization happens implicitly, but you can do it explicitly by calling the Windows [GetFullPathName() function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx).
+
+### Identifying the path
+
+The first step in path normalization is identifying the type of path. Paths fall into one of a few categories:
+
+- They are device paths; that is, they begin with two separators and a question mark or period (`\\?,` or `\\.`).
+- They are UNC paths; that is, they begin with two separators without a question mark or period. 
+- They are fully qualified DOS paths; that is, they begin with a drive letter, a volume separator, and a component separator (`C:\`).
+- They designate a legacy device (`CON`, `LPT1`).
+- They are relative to the root of the current drive; that is, they begin with a single component separator (`\`).
+- They are relative to the current directory of a specified drive; that is, they begin with a drive letter, a volume separator, and no component separator (`C:`).
+- They are relative to the current directory; that is, they begin with anything else (`temp\testfile.txt`).
+
+The type of the path determines whether or not a current directory is applied in some way. It also determines what the "root" of the path is.
+
+### Handling legacy devices
+
+If the path is a legacy DOS device such as `CON`, `COM1`, or `LPT1`, it is converted into a device path by prepending `\\.\` and returned.
+
+### Applying the current directory
+
+If a path isn't fully qualified, Windows applies the current directory to it. UNCs and device paths do not have the current directory applied. Neither does a full drive with separator C:\.
+
+If the path starts with a single component separator, the drive from the current directory is applied. For example, if the file path is `\utilities` and the current directory is `C:\temp\`, normalization produces `C:\utilities`.
+
+If the path starts with a drive letter, volume separator, and no component separator, the last current directory set from the command shell for the specified drive is applied. If the last current directory was not set, the drive alone is applied. For example, if the file path is `D:sources`, the current directory is `C:\Documents\`, and the last current directory on drive D: was `D:\sources\`, the result is `D:\sources\sources`. These "drive relative" paths are a common source of program and script logic errors. Assuming that a path beginning with a letter and a colon isn't relative is obviously not correct.
+
+If the path starts with something other than a separator, the current drive and current directory are applied. For example, if the path is `filecompare` and the current directory is `C:\utilities\`, the result is `C:\utilities\filecompare\`.
+
+> [!IMPORTANT]
+> Relative paths are dangerous in multithreaded applications (that is, most applications) because the current directory is a per-process setting. Any thread can change the current directory at any time. 
```

Comment:
We've added a new API for resolving paths to 2.1 that works around this issue. `Path.GetFullPath(string path, string basePath)` allows you to specify the "current directory" (basePath) you want to resolve against.